### PR TITLE
Add support of setting revisionHistoryLimit

### PR DESCRIPTION
**Pull request by MonkeyCanCode**
# Description

The current Helm chart doesn’t allow customization of the revisionHistoryLimit parameter, which is crucial for controlling the number of ReplicaSets retained for K8S deployment. This PR introduces the ability to set revisionHistoryLimit in the values file, defaulting to the standard value if not specified.

Also, I cleanup one comment in ingress.yaml which is not needed.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Documentation update
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

When revisionHistoryLimit is not set:
```sh
xxx@DESKTOP:~/k8s/polaris/helm/polaris(add_revisionHistoryLimit)$ cat myvalues.yaml | grep revision
revisionHistoryLimit: ~
xxx@DESKTOP:~/k8s/polaris/helm/polaris(add_revisionHistoryLimit)$ helm template polaris -f myvalues.yaml . | grep revision
walk.go:74: found symbolic link in path: /home/yong/k8s/polaris/helm/polaris/LICENSE resolves to /home/yong/k8s/polaris/LICENSE. Contents of linked file included and used
```
When revisionHistoryLimit is set to 0:
```
xxx@DESKTOP:~/k8s/polaris/helm/polaris(add_revisionHistoryLimit)$ cat myvalues.yaml | grep revision
revisionHistoryLimit: 0
xxx@DESKTOP:~/k8s/polaris/helm/polaris(add_revisionHistoryLimit)$ helm template polaris -f myvalues.yaml . | grep revision
walk.go:74: found symbolic link in path: /home/yong/k8s/polaris/helm/polaris/LICENSE resolves to /home/yong/k8s/polaris/LICENSE. Contents of linked file included and used
  revisionHistoryLimit: 0
```
When revisionHistoryLimit is set to 1:
```sh
xxx@DESKTOP:~/k8s/polaris/helm/polaris(add_revisionHistoryLimit)$ cat myvalues.yaml | grep revision
revisionHistoryLimit: 1
xxx@DESKTOP:~/k8s/polaris/helm/polaris(add_revisionHistoryLimit)$ helm template polaris -f myvalues.yaml . | grep revision
walk.go:74: found symbolic link in path: /home/yong/k8s/polaris/helm/polaris/LICENSE resolves to /home/yong/k8s/polaris/LICENSE. Contents of linked file included and used
  revisionHistoryLimit: 1
```

# Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by adutra — APPROVED** [review]

File: helm/polaris/templates/deployment.yaml [hunk 1]
```diff
@@ -31,6 +31,9 @@ spec:
   {{- if not .Values.autoscaling.enabled }}
   replicas: {{ .Values.replicaCount }}
   {{- end }}
+  {{- if not (has (quote .Values.revisionHistoryLimit) (list "" (quote ""))) }}
```
Comment (adutra) [inline helm/polaris/templates/deployment.yaml:34]: @MonkeyCanCode we just merged support for helm-unit. If you update your branch, maybe you could add a small unit test for this field? Especially for the logic around differentiating `0` from `nil` or `""`. Thanks 🙏

Comment (MonkeyCanCode) [reply helm/polaris/templates/deployment.yaml:34]: Nice. Will look into this later today.

Comment (MonkeyCanCode) [reply helm/polaris/templates/deployment.yaml:34]: @adutra this had being added. I really like this approach with validating helm (compared to how I was doing it earlier via helm template). I can add the remaining test cases in for the next PR if there is no concern on your end and no one is already working on it.

Comment (adutra) [reply helm/polaris/templates/deployment.yaml:34]: Excellent tests 👍 
And OK to add missing test cases little by little.

**Review by adutra — APPROVED** [review]

**Comment by MonkeyCanCode** [comment]
@adutra new PR for the remaining manifest had being created in https://github.com/apache/polaris/pull/409. Please take a look when you get a chance and let me know if anything should be changed.

**Comment by adutra** [comment]
@MonkeyCanCode I merged #409 before this one, which was probably a mistake. And now we have a conflict on this PR. Could you resolve the conflict please? I'll merge as soon as CI passes.

**Comment by MonkeyCanCode** [comment]
> @MonkeyCanCode I merged #409 before this one, which was probably a mistake. And now we have a conflict on this PR. Could you resolve the conflict please? I'll merge as soon as CI passes.

Checking

**Comment by MonkeyCanCode** [comment]
@adutra PR is fixed. What u did there is correct. The problem for the conflict is due to both MR making the same change to ingress.yaml (I fixed one thing there earlier for this PR...then one week later fix the same on the PR you just merged along with additional unit cases etc.)

**Review by adutra — APPROVED** [review]
